test/aave-v3/TestSupplyVault.t.sol

```diff
@@ -646,6 +646,156 @@ contract TestSupplyVault is TestSetupVaults {
         assertApproxEqAbs(uint256(userReward1_1), userReward1_2, 1);
     }
 
+    function testTransfer() public {
+        uint256 amount = 1e6 ether;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+
+        uint256 balance = daiSupplyVault.balanceOf(address(supplier1));
+        vm.prank(address(supplier1));
+        daiSupplyVault.transfer(address(supplier2), balance);
+
+        assertEq(daiSupplyVault.balanceOf(address(supplier1)), 0);
+        assertEq(daiSupplyVault.balanceOf(address(supplier2)), balance);
+    }
+
+    function testTransferFrom() public {
+        uint256 amount = 1e6 ether;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+
+        uint256 balance = daiSupplyVault.balanceOf(address(supplier1));
+        vm.prank(address(supplier1));
+        daiSupplyVault.approve(address(supplier3), balance);
+
+        vm.prank(address(supplier3));
+        daiSupplyVault.transferFrom(address(supplier1), address(supplier2), balance);
+
+        assertEq(daiSupplyVault.balanceOf(address(supplier1)), 0);
+        assertEq(daiSupplyVault.balanceOf(address(supplier2)), balance);
+    }
+
+    function testTransferAccrueRewards() public {
+        uint256 amount = 1e6 ether;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+
+        vm.warp(block.timestamp + 10 days);
+
+        uint256 balance = daiSupplyVault.balanceOf(address(supplier1));
+        vm.prank(address(supplier1));
+        daiSupplyVault.transfer(address(supplier2), balance);
+
+        uint256 expectedIndex = ERC20(rewardToken).balanceOf(address(daiSupplyVault)).rayDiv(
+            daiSupplyVault.totalSupply()
+        );
+        uint256 rewardsIndex = daiSupplyVault.rewardsIndex(rewardToken);
+        assertEq(expectedIndex, rewardsIndex);
+
+        (uint256 index1, uint256 unclaimed1) = daiSupplyVault.userRewards(
+            rewardToken,
+            address(supplier1)
+        );
+        assertEq(index1, rewardsIndex);
+        assertEq(unclaimed1, 0);
```

Comment:
it should be greater than 0 no? Since `supplier1` has accrued rewards for 10 days.

Reply:
But rewards were not claimed yet! So indexes are 0 and unclaimed amounts 0 (if I'm not mistaken)
should the rewards index take into account the underlying pool's liquidity mining rate? Or just be based on the reward token's balance of the vault (as it is now)?

Reply:
Because the index and unclaimed is fetched from storage, @Rubilmax is correct. If we want a view only getter, it must be from a separate external getter.

Reply:
Discovered the issue and fixed it. A lot of the tests were actually faulty because the test setup for aave v3 slightly differs from compound's setup, in that the suppliers are not reassigned to addresses properly. I fixed this, and also changed up some of compound's test code to make things more consistent.

Reply:
Ok so I think we should question more the tests when something is not clear